### ISSUE TITLE
Enhancement with error messages when TabbedTranslationAdmin is used

### ISSUE THREAD

**Issue by gabriel-petz** [deschler/django-modeltranslation#571]
I have some models on my projects that contains the modeltranslation implementation. On those models the field name is the one who applies the translation, so I have the additional fields called 'name_en' and 'name_pt_br', both fields are required, where null=False and blank=False, based on the original field. 

**The problem:**
On my admin I had implemented the TabbedTranslationAdmin as you can see on the following image. 
![image](https://user-images.githubusercontent.com/30831468/94498351-46f1a900-01d0-11eb-90d5-ed92d0512a60.png)

But, when the user doesn't fill all the fields like 'name_pt_br' and tries to save the object, the django admin returns the error saying to the user that he must correct the error bellow. But the TabbedTranslationAdmin doesn't show where is the error, making the user think the system have a bug on this page. 

![image](https://user-images.githubusercontent.com/30831468/94498461-989a3380-01d0-11eb-957f-c8884dcace84.png)
(The empty field example)

![image](https://user-images.githubusercontent.com/30831468/94498557-da2ade80-01d0-11eb-99c4-a5cb96253fef.png)
(Admin error message)

On that example above is easy to check where the problem is, but on a page with more fields using TabbedTranslationAdmin start to be a hard task to find out where is the empty field.

Now using the image bellow as example, imagine if the user doesn't fill the 'breadcrumb_pt_br' field and tries to save the object. The error still the same, but where is the empty field? 

![image](https://user-images.githubusercontent.com/30831468/94498850-9684a480-01d1-11eb-9326-4d3bcfa6b07e.png)
![image](https://user-images.githubusercontent.com/30831468/94499256-93d67f00-01d2-11eb-81c5-9f0fb2978df7.png)
(More advanced page with many tabs and error when the user tries to save with empty fields)

To find out where is the empty field the user must check all tabs, and that's a problem. 

My sugestion to improve the user experience is to highlight the tab that contains the error, like this:
![image](https://user-images.githubusercontent.com/30831468/94500213-1c561f00-01d5-11eb-8f61-61d9c73314aa.png)

Other suggestion is to change to the tab that contains the error: 
![image](https://user-images.githubusercontent.com/30831468/94500286-56272580-01d5-11eb-97e9-f416b70922a4.png)

**Comment by last-partizan** [comment]
Please check branch dev-2020 and report if it works.

**Comment by gabriel-petz** [comment]
I saw the changes on that branch, but this branch it's actually not workning


![](https://cdn.discordapp.com/attachments/692386071719182366/761234691377463316/unknown.png)

**Comment by marcoaurelioneto** [comment]
Hello @gabriel-petz and @last-partizan,

I was able to test it on branch 0.13.x. I replaced the tabbed_translation_fields.js file on branch 0.13.x with the branch dev-2020.

Apparently the expected behavior is correct. But we still have a bug.

The following code always uses the number of available tabs and always positions the last tab. But the error may not be there.

```
if (container.hasClass("errors"))
    activeTab = tabsList.length;
```

We need an index to find the tab that has the error. Something like:
```
$.each(groupedTranslations, function (groupId, lang) {
   ...
   var index = 0;
   $.each(lang, function (lang, el) {
        var $container = $(el).closest('td'),
         ...
        tabsContainer.append($panel);
        if (container.hasClass("errors"))
            activeTab = index;
        index++;
     });
     insertionPoint.el[insertionPoint.insert](tabsContainer);
     tabsContainer.tabs({
        active: activeTab,
     });
     ....   
```
If you prefer I can make a pull request for this correction.
Let me know which branch we can send this.

Regards
Marco Neto

**Comment by last-partizan** [comment]
@gabriel-petz yeah, install was broken in editable mode, now it's fixed, try it again please.

@marcoaurelioneto Hi, are you sure it always uses last tab? for me it working correctly (at least in simple tabs, i have not checked inlines).

Here is my idea of how my code should work:
- activeTab is 0 by default
- then we add tab, and check if it has errors.
- if it has errors - activeTabs is `tabsList.length`, which is same as index. We're adding tabs to list, so we don't have to count it by using index variable, we can get counter from list.

**Comment by marcoaurelioneto** [comment]
Hello, @last-partizan 

I ran the dev-2020 version and now it has no more errors.

As for the bug, to simulate the specific bug I followed the steps.

1) In my case, I have two tabs (1 - en and 2 - pt_br)
2) I edited only the second tab (2 - pt_br) and saved
3) As a result we have the following image

![image](https://user-images.githubusercontent.com/557439/94924880-b7305100-0494-11eb-989e-69389e044330.png)

Note that the system returns and is in tab 2 - en_br. But the error is in tab 1 - pt (see error in tab-1 in following image)

![image](https://user-images.githubusercontent.com/557439/94924944-d202c580-0494-11eb-802f-30c69e0bfe32.png)
